### ISSUE TITLE
Function `SimulatedEffect msg -> Cmd msg`?

### ISSUE THREAD

**Issue by jwoudenberg** [avh4/elm-program-test#100]
Way I would imagine using it: write my entire application using `SimulatedEffect` directly and save the trouble of having to create a custom `Effect` type.

Is the reason one doesn't exist technical, to encourage a certain way of writing tests, or something else?

**Comment by avh4** [comment]
elm-testable (the prototype for elm-program-test) originally worked that way.  I decided to try a new approach because that way seemed to make the production code a bit more unnatural, and I thought it was better to put more of the unnaturalness on the test code side than on the production code side.  Also that would make production applications have a transitive dependency on elm-test unless SimulatedEffect were moved to a separate package, which would complicate the discoverability of the API.

The real solution here is to modify elm-test (ideally with support from the core libraries, though that's not absolutely required) to allow direct inspection of Cmds so that SimulatedEffect and Effects are no longer needed.
